File: packages/std/src/parse-json.ts
/**
 * A0 stdlib: parse.json
 */
import type { StdlibFn, A0Record, A0Value } from "@a0/core";

export const parseJsonFn: StdlibFn = {
  name: "parse.json",
  execute(args: A0Record): A0Value {
    const input = args["in"];
    if (typeof input !== "string") {
      return { err: { code: "E_TYPE", message: "parse.json requires 'in' to be a string." } };
    }
    try {
      return JSON.parse(input) as A0Value;
    } catch (e) {
      const msg = e instanceof Error ? e.message : String(e);
      return { err: { code: "E_JSON", message: msg } };
    }
  },
};

File: go/pkg/stdlib/parse_json.go
package stdlib

import (
	"encoding/json"
	"fmt"

	"github.com/thomasrohde/agent0/go/pkg/evaluator"
)

// parse.json { in: string } → any
func stdlibParseJSON(args *evaluator.A0Record) (evaluator.A0Value, error) {
	input, _ := args.Get("in")
	inStr, ok := input.(evaluator.A0String)
	if !ok {
		return nil, fmt.Errorf("parse.json requires 'in' to be a string")
	}

	result, err := evaluator.ParseJSONToValue(json.RawMessage(inStr.Value))
	if err != nil {
		return nil, fmt.Errorf("%s", err.Error())
	}
	return result, nil
}
